image.go

```diff
@@ -38,6 +38,9 @@ type Image interface {
 	SetEntrypoint(...string) error
 	SetWorkingDir(string) error
 	SetCmd(...string) error
+	SetOS(string) error
+	SetOSVersion(string) error
+	SetArchitecture(string) error
```

Comment:
Eventually I feel like these options make more sense as options on `NewImage`. Having constructor options instead of a setter would convey that is not valid to change these values for any image other than an empty image. But I can see how the awkwardness of our existing `FromBase` option makes that difficult to support without broader changes.

Reply:
Yeah, good call. I was also wondering how it will intersect with the needs of https://github.com/buildpacks/imgutil/issues/54 which would also need very similar looking flags but to instead filter the `FromBaseImage` platform.